torch/nested/_internal/ops.py

```diff
@@ -884,16 +884,91 @@ def split_with_sizes_default(func, *args, **kwargs):
+        start_val, length_val = new_kwargs["start"], new_kwargs["length"]
+        end_val = start_val + length_val
+        batch = inp._offsets.shape[0] - 1
+        if end_val > batch:
+            raise RuntimeError(
+                f"narrow(): start ({start_val}) + length ({length_val}) "
+                f"exceeds dimension size ({batch})"
+            )
+
+        # clamp start, end values
+        if start_val < 0:
+            start_val += inp._values.size(dim)
+        if end_val < 0:
+            end_val += inp._values.size(dim)
+        start_val = max(min(start_val, inp._values.size(dim)), 0)
+        end_val = max(min(end_val, inp._values.size(dim)), 0)
+        length_val = max(min(length_val, end_val - start_val), 0)
+
+        # shortcut if no actual narrowing is happening; this helps us ensure
+        # that length < batch size if we don't take this path
+        if length_val == inp.size(0):
+            return inp.detach()
+
+        # +1 to include last offset. Also normalize offsets to start at 0.
+        out_kwargs["offsets"] = (
+            inp._offsets[start_val : start_val + length_val + 1]
+            - inp._offsets[start_val]
+        )
+        # metadata cache may no longer be accurate since offsets have changed
+        if "_metadata_cache" in out_kwargs:
+            del out_kwargs["_metadata_cache"]
+
+        if inp._lengths is not None:
+            out_kwargs["lengths"] = inp._lengths[start_val : start_val + length_val]
+
+        # unbacked SymInt for new storage offset
+        new_storage_offset = (
+            inp._values.storage_offset()
+            + (inp._offsets[start_val] * inp._values.stride(dim))
+        ).item()
+        torch._check_is_size(new_storage_offset)
+
+        # compute symbolic start involving unbacked SymInt
+        start = (
+            new_storage_offset - inp._values.storage_offset()
+        ) // inp._values.stride(dim)
+        torch._check_is_size(start)
+        torch._check(start <= inp._values.size(dim))
+
+        # unbacked SymInt for length
+        length = (inp._offsets[start_val + length_val] - inp._offsets[start_val]).item()
+        torch._check_is_size(length)
+        # we can say this because we short-circuit earlier if length == inp._values.size(dim)
+        torch._check(length < inp._values.size(dim))
+        torch._check(start + length <= inp._values.size(dim))
+
+        # compute new sizes / strides from symbolic values
+        new_sizes = list(inp._values.size())
+        new_sizes[dim] = length
+        new_strides = list(inp._values.stride())
+
+        # apply view with new sizes / strides / storage offset
+        new_values = inp._values.as_strided(new_sizes, new_strides, new_storage_offset)
+        return NestedTensor(new_values, **out_kwargs)
+
+    if inp._lengths is not None or inp._ragged_idx != 1:
+        raise RuntimeError(
+            "narrow(): not yet supported for non-contiguous nested tensors on dim != 0"
```

Comment:
maybe its my own fault, but I keep reading this as "tensors that are not contiguous on dim != 0" 
If it were instead worded as 
`not yet supported on dim != 0 for non-contiguous nested tensors` I would not make that mistake.

Reply:
much clearer! fixed